core/src/main/java/nl/nn/adapterframework/http/WebServiceSender.java

```diff
@@ -132,19 +132,20 @@ protected HttpRequestBase getMethod(URI uri, Message message, ParameterValueList
 		}
 		if (log.isDebugEnabled()) log.debug(getLogPrefix()+"SOAPMSG [" + soapmsg + "]");
 
-		HttpRequestBase method = super.getMethod(uri, new Message(soapmsg), parameters, session);
+		HttpRequestBase method = super.getMethod(uri, soapmsg, parameters, session);
 		log.debug(getLogPrefix()+"setting SOAPAction header ["+soapActionURI+"]");
 		method.setHeader("SOAPAction", soapActionURI);
 		return method;
 	}
 
 	@Override
 	protected Message extractResult(HttpResponseHandler responseHandler, IPipeLineSession session) throws SenderException, IOException {
-		String httpResult = null;
+		Message httpResult = null;
 		try {
-			httpResult = super.extractResult(responseHandler, session).asString();
+			httpResult = super.extractResult(responseHandler, session);
+			httpResult.preserve();
 		} catch (SenderException e) {
-			soapWrapper.checkForSoapFault(getResponseBodyAsString(responseHandler), e);
+			soapWrapper.checkForSoapFault(getResponseBody(responseHandler), e);
```

Comment:
Gaat dit wel goed? `getResponseBody` kan een `java.io.IOException: Stream closed` exceptie gooien bij het lezen van een foutive response.
In `getResponseBodyAsString` zit hiervoor een try/catch die silent `null` terug geeft. Nu krijg je een exceptie in een exceptie